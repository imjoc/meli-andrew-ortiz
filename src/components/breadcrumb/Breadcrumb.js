import React, { Component } from "react";

if (process.env.WEBPACK) require("./breadcrumb.scss");

class Breadcrumb extends Component {

	render () {
		const data = this.props.data || [];

		return (<div className="container-breadcrumb">
			<ul className="breadcrumb">
				{data.map((row, index) => {
					return (
						<li key={index} onClick={() => {
							this.props.baseLink(row);
						}}>{row}</li>
					);
				})}
			</ul>
		</div>);
	}
}

export default Breadcrumb;