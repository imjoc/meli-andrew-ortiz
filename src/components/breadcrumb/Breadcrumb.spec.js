import React from "react";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import Breadcrumb from "./index";
configure({ adapter: new Adapter() });

const categories = ["Tecnelogias", "Macbook"];

it("renders Breadcumb component correctly", function() {
    const tree = renderer.create(
        <Breadcrumb data={categories}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});


test("Check render categories in Breadcrumb", function() {
    const component = shallow(
        <Breadcrumb data={categories}/>
    );

    expect(component.text()).toEqual("TecnelogiasMacbook");
});
