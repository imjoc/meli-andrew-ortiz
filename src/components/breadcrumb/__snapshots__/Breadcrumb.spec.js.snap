// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Breadcumb component correctly 1`] = `
<div
  className="container-breadcrumb"
>
  <ul
    className="breadcrumb"
  >
    <li
      onClick={[Function]}
    >
      Tecnelogias
    </li>
    <li
      onClick={[Function]}
    >
      Macbook
    </li>
  </ul>
</div>
`;
